### PR TITLE
Update new build

### PR DESCRIPTION
**Pull request by tier940**
- CEuから登場した新しいビルドシステムを導入
- 起動確認もクリア済み

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,101 +1,154 @@
...
+# Username is set with the 'MAVEN_USER' environment variable, default to "NONE"
+# Password is set with the 'MAVEN_PASSWORD' environment variable, default to "NONE"
+customMavenPublishUrl =
+
+# Enable spotless checks
+# Enforces code formatting on your source code
+# By default this will use the files found here: https://github.com/GregTechCEu/Buildscripts/tree/master/spotless
+# to format your code. However, you can create your own version of these files and place them in your project's
+# root directory to apply your own formatting options instead.
+enableSpotless = false
```
Comment (miozune) [inline gradle.properties:134]: メモ：進行中のPRが全てマージされるまでは有効にしないでください(一応rebaseすれば大丈夫ですが面倒なので)

File: repositories.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Add any additional repositories for your dependencies here
+
+repositories {
+    maven { // MixinBooter
```
Comment (miozune) [inline repositories.gradle:4]: `includeWellKnownRepositories`がtrueなのでmodmaven以外は不要です

File: gradle.properties [hunk 3]
```diff
@@ -1,101 +1,154 @@
...
+# restart Minecraft in development. Choose this dependent on your mod:
+# Do you need consistent player progressing (for example Thaumcraft)? -> Select a name
+# Do you need to test how your custom blocks interacts with a player that is not the owner? -> leave name empty
+# Alternatively this can be set with the 'DEV_USERNAME' environment variable.
+developmentEnvironmentUserName = Developer
+
+# Enables using modern java syntax (up to version 17) via Jabel, while still targeting JVM 8.
+# See https://github.com/bsideup/jabel for details on how this works.
+# Using this requires that you use a Java 17 JDK for development.
+enableModernJavaSyntax = true
```
Comment (miozune) [inline gradle.properties:31]: メモ：進行中のPRが全てマージされたらソースコードのJava17移行(instanceofマッチングなど)を一括でやりましょう

File: dependencies.gradle [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    // Debug Draconic Evolution/Additions
+    compileOnlyApi rfg.deobf("curse.maven:redstone-flux-270789:2920436")
+    compileOnlyApi rfg.deobf("curse.maven:brandons-core-231382:3408276")
+    compileOnlyApi rfg.deobf("curse.maven:draconic-evolution-223565:3431261")
+    compileOnlyApi rfg.deobf("curse.maven:draconicadditions-314515:3514704")
+    if (project.debug_de.toBoolean()) {
+        implementation rfg.deobf("curse.maven:redstone-flux-270789:2920436")
+        implementation rfg.deobf("curse.maven:brandons-core-231382:3408276")
+        implementation rfg.deobf("curse.maven:draconic-evolution-223565:3431261")
+        implementation rfg.deobf("curse.maven:draconicadditions-314515:3514704")
```
Comment (miozune) [inline dependencies.gradle:61]: `implementation`=`compileOnly`+`runtimeOnly`なのでこの4つは`runtimeOnly`でいいです
https://gtnh.miraheze.org/wiki/Development#Dependency_types

File: dependencies.gradle [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    implementation rfg.deobf("curse.maven:top-245211:2667280")
+    implementation rfg.deobf("curse.maven:ctm-267602:2915363")
+
+    // use a local jar for GroovyScript to avoid a bug from Mixin and ForgeGradle,
+    // causing crashes at run-time when deobfuscated
+    implementation files("libs/groovyscript-0.4.0.jar")
+    //implementation rfg.deobf("curse.maven:groovyscript-687577:4399621")
+
+    // Debug Draconic Evolution/Additions
+    compileOnlyApi rfg.deobf("curse.maven:redstone-flux-270789:2920436")
```
Comment (miozune) [inline dependencies.gradle:53]: どうせこのmodに依存するmodが無いので`compileOnly`でいいです

File: dependencies.gradle [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    implementation rfg.deobf("curse.maven:ender-io-conduits-304346:3328808")
+    implementation rfg.deobf("curse.maven:endercore-231868:2972849")
+    implementation "info.loenwind.autoconfig:AutoConfig:1.12.2-1.0.2"
+    implementation "info.loenwind.autosave:AutoSave:1.12.2-1.0.11"
+    implementation rfg.deobf("curse.maven:baubles-227083:2518667")
+    implementation rfg.deobf("curse.maven:shadowfacts-forgelin-248453:2785465")
+    implementation rfg.deobf("curse.maven:ae-additions-extra-cells-2-fork-493962:3814371")
+
+    // Soft Dependencies
+    compileOnlyApi rfg.deobf("curse.maven:gregtech-food-option-477021:4555292")
```
Comment (miozune) [inline dependencies.gradle:41]: 同上

File: gradle.properties [hunk 7]
```diff
@@ -1,101 +1,154 @@
+modName = GTExportCore
```
Comment (miozune) [inline gradle.properties:1]: `Export` :concern:
